changelog.d/979.fix.md

```diff
@@ -0,0 +1 @@
+`to_unix_timestamp`, `to_float`, and `uuid_v7` can now return an error if the supplied timestamp is unrepresentable as a nanosecond timestamp. Previously the function calls would panic.
```

Comment:
Should this file be `.breaking.`?

Reply:
Yes, indeed. 

💭 This PR makes me think... we probably want to advertise a policy about breaking changes in VRL. In this case we are fixing a panic at the cost of requiring users to usage of `to_unix_timestamp` their VRL scripts. IMO this is worth the effort. 

In the future, we could do something like:
* in release X+1 we announce our intention to change behavior
* release X+2 we actually include the change 

On the other hand, this might be too much process for smaller changes.

P.S. This would be a non-breaking change if we finally got to introducing a non-strict mode.

Reply:
Agreed, it'd be nice to have a VRL deprecation process for riskier breaking changes that we aren't sure about to let users give us feedback. In this case I think it's ok to make the breaking change since the migration is simple (add a `!` to preserve the existing behavior).